### ISSUE TITLE
DOSBox Staging 0.81 release steering ticket

### ISSUE THREAD

**Issue by Grounded0** [dosbox-staging/dosbox-staging#3091]
Modified: Wed 14 Feb 2024 05:07:40 UTC

## Required actions to clear for release

@johnnovak responding to me:

### Release Candidate 1 (release date: 22 Dec 2023)

- [x] The final release notes are available, so early-adopters of the RC are informed about the new features

### Release (release date: 14 Feb 2024)

- [x] Total code freeze until the final version is out—only regression fixes are allowed in this period and maybe minor non-regression bug fixes, but that very sparingly
- [x] The dev team also starts vigorously testing the RC in parallel with the community

None of the above has happened yet, all that is aimed for Dec 2023.

### 0.81.0 Release Candidate

https://dosbox-staging.github.io/releases/release-notes/0.81.0-rc/

### 0.81.0 Release

https://dosbox-staging.github.io/releases/release-notes/0.81.0/

### Code of Conduct & Contributing Guidelines

- [X] Yes, I agree.

**Comment by johnnovak** [comment]
Nice one @Grounded0 🎉

**Comment by Grounded0** [comment]
@johnnovak on the current state of the release notes:

You can view the current release draft here:
https://github.com/dosbox-staging/dosbox-staging/releases

I will write the notes for my own major features myself, so please leave those alone (adaptive CRT shaders, VGA double-scanning improvements, MIXER improvements, MIDI muting, image capture enhancements, plus maybe a few other things I can't remember). In fact, most of them are already written, I just need to "refactor" my PR descriptions and reformulate them for non-devs.

**Comment by Grounded0** [comment]
I can write much of the bits left outside those major features myself but some code contributors may want to write parts they authored themselves so that can be figured out later.

**Comment by FeralChild64** [comment]
The changes I did (or ported form DOSBox-X, or finished someone else work) so far:

- DOS commands/programs improvements
  - implemented `TREE.COM` and `SETVER.EXE`
  - `MORE.COM` extended with features from modern Windows
  - `LS.COM` now accepts multiple paths / patterns and ignores hidden / system files, unless /a switch is given
  - `DIR /p` now uses `MORE` engine for display
  - it is now possible to disable` COMMAND.COM` shortcuts to manage the configuration
- all file / directory DOS attributes are now supported; on Linux / macOS they are stored as extended file attributes, in a format compatible with WINE
- maximum XMS memory size increased to 3GB
- rewritten keyboard hardware emulation, sporadic problem with keyboard not reacting any more after exiting Windows 3.11 for Workgroups should be fixed
- implemented register-level emulation of PS/2 mice
- Microsoft IntelliMouse Explorer (5 buttons + wheel) mouse is now emulated
- it is now possible to disable built-in DOS mouse driver
- on multi-display hosts it is now possible to release captured mouse or use seamless mouse integration in fullscreen mode
- `CONFIG.COM -wc` does not eat away empty lines from [autoexec] section any more
- implemented UTF-8 encoding support in [autoexec] section of the configuration file
- size limitation of [autoexec] section is now gone
- HP Vectra fast A20 handling is now emulated (MS-DOS `HIMEM.SYS` switch `/machine:14`)
- fixed mouse support in MS Word 5.5 for DOS
- fixed problems with internal mouse driver and certain screen modes above 80 columns

There is also a high chance we'll see a VirtualBox mouse protocol supported - PR https://github.com/dosbox-staging/dosbox-staging/pull/3089

**Comment by Grounded0** [comment]
Added UTC timestamp to reduce disambiguity.

**Comment by Grounded0** [comment]
I've finally got a 10 A4 page printout of all the PRs. Wow.

I've thought that this document structure would serve us well with the release notes:

MAJOR NEW FEATURES

Handled by @johnnovak and @kcgen 

GRAPHICS

Handled by @Grounded0 

10 picks documented by me

Collapsible list of all PRs sorted to this category

SOUND

Handled by @Grounded0 

10 picks documented by me

Collapsible list of all PRs sorted to this category

INPUT

Handled by @Grounded0 

10 picks documented by me

Collapsible list of all PRs sorted to this category

HOST INTEGRATION

Handled by @Grounded0 

10 picks documented by me

Collapsible list of all PRs sorted to this category

MISCELLANEOUS

Handled by @Grounded0 

10 picks documented by me

Collapsible list of all PRs sorted to this category

CONTRIBUTORS

Handled by @kcgen

**Comment by Grounded0** [comment]
Personal progress list:

- [ ] Sort tickets into 6 categories (done by 12 Nov 2023)
- [ ] GRAPHICS (done on 15 Nov 2023)
- [ ] SOUND (done on 18 Nov 2023)
- [ ] INPUT (done on 21 Nov 2023)
- [ ] HOST INTEGRATION (done on 24 Nov 2023)
- [ ] MISCELLANEOUS (done on 27 Nov 2023)

**Comment by johnnovak** [comment]
@Grounded0 @kcgen @FeralChild64 @weirddan455 

I've done a pass through all our tickets and tagged them appropriately & synced the release drafter to our updated/new tag names.

You can view the current release draft here:
https://github.com/dosbox-staging/dosbox-staging/releases

**IMPORTANT:** **Make sure to click on the "v0.81.0" heading to view the full release notes as the "Releases" page only displays a shortened version of it.**

So now it's a good time to start sorting the items into categories from the release notes draft @Grounded0 

I will handle the below tickets myself (90%+ of these are mine) — no action is needed from anyone on these.

Everybody else, please post your release notes here as comments. Don't create multiple comments, always update a _single_ comment that contains all your contribution to the release notes so I won't go mad when trying to collate them all 😄 Thanks!

---

Stuff I'll be dealing with:

## General video

- #2456
- #2503
- #2769 

## VGA emulation

- #2398 (@kcgen's work which I refined in the below tickets, so I'm happy to write the notes for the whole thing myself)
- #2283
- #2284
- #2282 
- #2654
- #2570 
- #2583 
- #2728 (internal stuff; ignore)
- #2393 (got added then later removed; ignore)
- #2373 (got changed then later refined/changed; ignore)

## Shaders

- #2777
- #2819
- #3083
- #2442 
- #2866 (internal; ignore)

## Audio

- #2868
- #2228
- #3105

## MIDI

- #2308
- #2318
- #3011
- #2591 (@weirddan455's, but related to the previous ticket)
- #3105

## MIDI

- #2308
- #2318
- #3011
- #2591 (@weirddan455's, but related to the previous ticket)

**Comment by Grounded0** [comment]
Can i continue to use this draft dated 8 Nov 02:44 UTC saved as PDF by @kcgen. I don't have permissions to view that.

**Comment by johnnovak** [comment]
> @johnnovak
> 
> Can i continue to use this draft dated 8 Nov 2023 02:44 UTC saved as PDF by @kcgen? I don't have permissions to view that.

@Grounded0 No, because I went through all the items and have added missing tags a few hours ago today. Please use this PDF then:

[release-notes-draft-v0.81.0-2023-11-13T015-08.pdf](https://github.com/dosbox-staging/dosbox-staging/files/13330920/release-notes-draft-v0.81.0-2023-11-13T015-08.pdf)

Sorry if this is causing you some re-work. We should make the initial combing through all the merged PRs and adding missing tags the first step before any further work is done with the release notes. I'll add this to the release process doco. It's just I was on holidays and did not have time to do this pass earlier.

**Comment by Grounded0** [comment]
It didn't since i was slacking off and watching Game Developers Conference Classic Game Postmortems. Time to work! 🪓

**Comment by johnnovak** [comment]
> It didn't since i was slacking off and watching Game Developers Conference Classic Game Postmortems. Time to work! 🪓

Kewl, hope it was fun to watch 😄 Btw, I think we should restore your privileges (or grant you new ones) so you can do the same things as the dev team if you're working on release notes and stuff. That would make sense, no @kcgen ?

**Comment by kcgen** [comment]
Yes; @Grounded0 has triage rights - I wasn't sure what's needed to see the draft notes (there's a discussion here):

https://github.com/orgs/community/discussions/4192

@Grounded0 - will bump to maintainer and see what happens :-)

**Comment by Grounded0** [comment]
Can see it now. Downsides: Have to be extremely careful where i click again on GitHub which was the reason i wanted to be demoted before.

**Comment by johnnovak** [comment]
> Can see it now. Downsides: Have to be extremely careful where i click again on GitHub which was the reason i wanted to be demoted before.

Huh? Coordination issue? Just don't delete anything 😄 If too risky, we can demote you again — up to you 😄

**Comment by weirddan455** [comment]
I wouldn't worry about it too much @Grounded0  I have seen @johnnovak and others here fat finger that "Close comment"/"Close PR" button.  They can be re-opened.

**Comment by Grounded0** [comment]
I'm now on page 10/17 and will crunch through the rest on Thu 16 Nov 2023.

**Comment by MasterO2** [comment]
**17 pages** of 0.81 features? Wow.

**Comment by Grounded0** [comment]
I'll fix duplicate entries when i wake up.

**Comment by Grounded0** [comment]
@johnnovak 

So you're leaving Voodoo stuff for me?

https://github.com/dosbox-staging/dosbox-staging/issues/3091#issuecomment-1807387965

**Comment by johnnovak** [comment]
> @johnnovak
> 
> So you're leaving Voodoo stuff for me?
> 
> [#3091 (comment)](https://github.com/dosbox-staging/dosbox-staging/issues/3091#issuecomment-1807387965)

You can have it 😎

**Comment by Grounded0** [comment]
I'll take it.

**Comment by Grounded0** [comment]
It has been agreed on IRC that @kcgen will take Voodoo to share workload more evenly.

**Comment by johnnovak** [comment]
> I've now finished my portion of the release notes.

Can you upload it as a GitHub gist and paste the link here? I'll do the rest.

**Comment by Grounded0** [comment]
https://github.com/dosbox-staging/dosbox-staging/issues/3091#issuecomment-1807227593

**Comment by johnnovak** [comment]
> @johnnovak
> 
> [#3091 (comment)](https://github.com/dosbox-staging/dosbox-staging/issues/3091#issuecomment-1807227593)

Cheers! 🍻

**Comment by FeralChild64** [comment]
Added one more feature to my list:
- it is now possible to select between modern and historic (DOS era) locale style

**Comment by Grounded0** [comment]
Are the release notes now ready by everyone so i tick the box for final release notes being ready?

**Comment by johnnovak** [comment]
> Are the release notes now ready by everyone so i can tick the box for final release notes being ready?

The release is aimed for 22 December. Working backwards, these are the key dates and milestones:

- Last code change gets merged by **10 Dec**, so **this Sunday**. Basically, only my video capture aspect ratio fix is outstanding. PR comes today, will get merged in a day or two.

- **10 Dec** is a soft-code-freeze. Internal testing starts by the dev team and contributors. We only merge code after that if we must fix serious regressions.

- Between **10 and 17 Dec** I aim to finalise the release notes. This includes writing my own stuff and prodding people to provide their part, or writing them myself if all fails 😅 The final release notes will be available on the preview website so we can all review it and make corrections if necessary until **21 Dec**.

- Until **21 Dec** translators have the chance to update their translations. Strictly no code changes, only updating resource files. 21 Dec because we still need to make a quick smoke test to make sure the release artifacts can be downloaded, they work, the installer works, etc.

- Release hits the shelves on **22 Dec** with a big fanfare 😎 Code freeze period for the next 6-10 weeks starts (roughly); we only fix regressions during this period and strictly no new features will be merged. The whole dev team does proactive testing during this period to find problems instead of working on new stuff.

---

Hope that clears up any question @Grounded0. After the release is out, I will create a document on the wiki with a formalised version of this process, then we can all refer to it going forward.

**Comment by FeralChild64** [comment]
I have just one PR in the queue, to make few strings translatable: https://github.com/dosbox-staging/dosbox-staging/pull/3178. I am not planning any more code changes for 0.81.0. The Polish translation is up to date at this moment, but will need a minor update once the mentioned PR is merged.

I'll, of course, review the draft release notes, but definitely not today.

**Comment by rderooy** [comment]
I have an updated Dutch translation ready. Just waiting on #3178 to land for the final fix-ups.

**Comment by kcgen** [comment]
@johnnovak, all looks good :+1: 

Just for ref: suggest following the release steps (starting w/ the RC) to get practice with the overall process: 

https://github.com/dosbox-staging/dosbox-staging/wiki/Maintenance-and-release-notes#6-releasing-a-new-version

Describes creating the new `release/0.81.x` branch, from which the official releases will evolve and be tagged from:
- RC tags: `v0.81.0-rc1`, `v0.81.0-rc2`, …
- final tag: `v0.81.0`, and
- bug-fix tags: `v0.81.1`, `v0.81.2`, …


Each release series is a branch hanging off the `main` line, just like our personal branches hang off `main`. The release branches are never rebased though -- instead, they are anchored back in time at the point they branched off main.

We then can keep adding more changes to a given release branch and ultimately tagging and release off the release branch, without any involvement from `main`.

**Comment by johnnovak** [comment]
Thanks for that @kcgen. Some of those steps are really not obvious and dead easy to forget. I'm a big fan of checklists and run sheets as I have a shit memory 😅

**Comment by kcgen** [comment]
I've augmented the list based on thing that weren't obvious to me (over the past handful of releases), and no doubt it will need more clarification and improvement with your fresh eyes, as well. Very happy to help and clarify!

**Comment by johnnovak** [comment]
> I've augmented the list based on thing that weren't obvious to me (over the past handful of releases), and no doubt it will need more clarification and improvement with your fresh eyes, as well. Very happy to help and clarify!

Cheers! I think it will be best if I cut the RC a couple of days before the actual release dat, say 3-4, so I have ample time to fix my mistakes and discuss things with you 😄

**Comment by johnnovak** [comment]
> I'll, of course, review the draft release notes, but definitely not today.

When you get to that, please also let me know if you want to highlight anything else input related (or anything else you worked on) in the notes in addition to @Grounded0's draft. Otherwise I might add a few more things I think are important

@kcgen do you want to provide a draft of the presentation mode stuff you worked on? I guess I could figure out what you exactly did, but you're closest to it and it's not exactly trivial to understand that stuff 😄 You've also changed the vsync defaulting logic if I remember correctly, so there two things sound quite important to me.

@Grounded0 the draft looks good and you've managed to describe things succinctly. Well done! One suggestion is to shorten the longer titles somewhat, so 4-5 words max. Very long titles just don't look good, so some need a bit of trimming 😄 ✂️ Another thing I'm planning to add to your notes is actual config snippet examples where applicable. So if you could help with that, that would be appreciated.

**Comment by Grounded0** [comment]
The reason it has been written the way its written is to not have everything listed but the most notable few from each category. If someone wants to add more entries they are free to do that but i'm probably not going to do it. Each category will have all the PRs from that category as a expandable list so everything is visible to someone who wants to inspect changes at a PR level.

**Comment by FeralChild64** [comment]
I understand. I consider the points 1. and 2. as "mandatory" (as these are fixes to the current descriptions), 3. as "IMHO nice to have", 4., 5., and 6. as "we can add these, but they are of low importance", and 7. as "I don’t know if we should really go that deep".

**Comment by johnnovak** [comment]
@FeralChild64 I agree with your suggestions, thanksm Having "too much" info is not a problem; people can just scroll past the parts they don't care about.

I will create the draft based on the current submissions on the preview website in the next few days, then you guys can suggest changes/additions/corrections to the final document.

**Comment by johnnovak** [comment]
@Grounded0 @kcgen @weirddan455 @kklobe @FeralChild64 @shermp et all.

Ok, so I'm done with the release notes. I'm in the process of adding screenshots, audio snippets, and maybe some videos today and maybe tomorrow, then I'll publish it to the _preview_ website once finished. Should come today, but **_no later than tomorrow_**, no matter what.

The URL in case someone doesn't know: https://dosbox-staging.github.io/preview/dev/

Then we can review it internally in the next few days, and make corrections if needed, etc. Maybe I'll have time to finish updating the getting started guide too before the RC (e.g., with info on the adaptive shaders). Regardless, the RC goes out on 22 Dec, then I'll finish up the guide during the RC period.

So excited for this, it's gonna be a HUGE release 😎 🤘🏻 

Wait until I show you my "marketing material" for the release... (a bunch of images people can share wherever they like).

**Comment by Grounded0** [comment]
I'm going to tick the box then.

**Comment by johnnovak** [comment]
@Grounded0 @kcgen @kklobe @weirddan455 @FeralChild64 @shermp 

Behold the 0.81.0-RC 😎 
https://dosbox-staging.github.io/preview/dev/downloads/release-notes/0.81.0-rc/

Some minor things outstanding for tomorrow because I ran out of steam:

- Add IMFC and FluidSynth portamento fix audio examples
- Mention the last-minute title bar changes

@Grounded0 I treated your contribution as raw material and reshuffled a few things, hope you don't mind. Your idea of presenting things per functional category (graphics, sound, input, etc.) was really good, so this is what we'll do going forward.

I wrote some Python scripts to categorise the PRs according to these functional categories so no one will need to do this manually ever again... I've also relabeled **all** PRs, so they're added to the correct categories automatically. I'll make that script part of the automatic release drafter process later because the current release drafter automation is a bit dumb, now I can only run it manually on my machine.

Comments are welcome and minor change suggestions and factual fixes, etc, but I won't fundamentally change anything—it took me a week to get the whole thing to this state 😅

**Comment by FeralChild64** [comment]
Nice! It must have been a lot of effort to describe this. I have just two remarks:

1.  _improved FAT filesystem integration_ -> _improved filesystem integration_; the host FS is probably not FAT nowadays

2. _Other mouse improvements_ chapter should mention the VirtualBox mouse protocol support, to be used with Windows 3.1x mouse driver by Javis Pedro: https://git.javispedro.com/cgit/vbmouse.git/about/

**Comment by kklobe** [comment]
Fantastic work @johnnovak! After going back to 0.80.1 and doing a bit more testing, I see that the vertical bars issue actually doesn't manifest on Rosetta x64, nor do I see it on Windows x64 for dynamic core (it does show on Normal core, just like the rounding fix). Maybe we could get another verification or two from testers first, then revise the wording a bit, e.g.:

```
 the issue does not manifest in 32-bit x86 builds, but it does in all non-Intel builds and some 64-bit x86 builds.
```

**Comment by Grounded0** [comment]
I think we agreed to eliminate dollar amounts from donations so those can be taken out. Just showing which bucket the donor fall into will be enough.

**Comment by Grounded0** [comment]
Also my friend Hannu Hartikainen donated at a 5 USD backer tier through GitHub Sponsors.

**Comment by johnnovak** [comment]
@Grounded0 I did export the expenses CSV and worked from that. Looks like I missed your friend.

Wasn't aware of the decision to remove aggregates but it also crossed my mind to maybe not do it.

Will fix these latet.

**Comment by johnnovak** [comment]
> Fantastic work @johnnovak! After going back to 0.80.1 and doing a bit more testing, I see that the vertical bars issue actually doesn't manifest on Rosetta x64, nor do I see it on Windows x64 for dynamic core (it does show on Normal core, just like the rounding fix). Maybe we could get another verification or two from testers first, then revise the wording a bit, e.g.:
> 
> ```
>  the issue does not manifest in 32-bit x86 builds, but it does in all non-Intel builds and some 64-bit x86 builds.
> ```

Thanks @kklobe. Please send me the revised text if you figure this out; I'm busy with the getting started guide now, then with cutting the release.

**Comment by Grounded0** [comment]
[GitHub: Viewing your sponsors and sponsorships](https://docs.github.com/en/sponsors/receiving-sponsorships-through-github-sponsors/viewing-your-sponsors-and-sponsorships)

**Comment by johnnovak** [comment]
@Grounded0 @FeralChild64 

Ok guys, I've addressed your comments, plus added more stuff to the notes. I've pretty much maxed this out now, I'm done 😅 Let me if you if you spot any errors, though.

Moving on to finalising the Getting Started guide; I want to release it with the RC because it could short-circuit a lot of questions. Then doing the build on Wednesday, probably.

**Comment by FeralChild64** [comment]
> DOSBox Staging has had experimental support for the VirtualBox mouse protocol for quite some time now

There was no VirtualBox mouse protocol support in the latest release, at all - only the VMWare one. And you don't need `mouse_capture` set to `seamless` for this driver, it works with `onclick` too.

**Comment by johnnovak** [comment]
> > DOSBox Staging has had experimental support for the VirtualBox mouse protocol for quite some time now
> 
> There was no VirtualBox mouse protocol support in the previous release, at all - only the VMWare one. And you don't need `mouse_capture` set to `seamless` for this driver, it works with `onclick` too.

Can you provide the full revised text @FeralChild64  I can just paste in? Thanks.

**Comment by FeralChild64** [comment]
Maybe something like this?

> DOSBox Staging has had experimental support for the VMware mouse protocol for quite some time now, and in this version we have added a VirtualBox mouse protocol. The primary motivation is to enable seamless mouse operation in Windows 3.x.
> 
> Just install [Javis Pedro’s Windows 3.x driver](https://git.javispedro.com/cgit/vbmouse.git/about/) and enjoy the seamless Windows 3.x experience!

**Comment by johnnovak** [comment]
> Maybe something like this?
> 
> > DOSBox Staging has had experimental support for the VMware mouse protocol for quite some time now, and in this version we have added a VirtualBox mouse protocol. The primary motivation is to enable seamless mouse operation in Windows 3.x.
> > Just install [Javis Pedro’s Windows 3.x driver](https://git.javispedro.com/cgit/vbmouse.git/about/) and enjoy the seamless Windows 3.x experience!

Done.

**Comment by Grounded0** [comment]
Now that the RC is out i ticked more boxes:

[x] Total code freeze until the final version is out—only regression fixes are allowed in this period and maybe minor non-regression bug fixes, but that very sparingly

[x] The dev team also starts vigorously testing the RC in parallell with the community

Hopefully everyone is now onboard with these two statements.

**Comment by Grounded0** [comment]
0.81.0-RC release notes:

https://dosbox-staging.github.io/releases/release-notes/0.81.0-rc/

**Comment by MasterO2** [comment]
I have posted about Dosbox-Staging 0.81 RC to the following places:

1. https://www.reddit.com/r/emulation/comments/18ofpyx/dosboxstaging_081_rc_has_now_been_released_with/
2. https://www.reddit.com/r/linux_gaming/comments/18ofras/dosboxstaging_081_rc_has_now_been_released_with/
3. https://www.reddit.com/r/dosgaming/comments/18ofsbs/dosboxstaging_081_rc_has_now_been_released_with/
4. https://www.reddit.com/r/dosbox/comments/18ofuqs/dosboxstaging_081_rc_has_now_been_released_with/
5. https://www.reddit.com/r/programming/comments/18oj7l0/dosboxstaging_0810rc_has_been_released_now_with/
6. https://www.reddit.com/r/carmageddon/comments/18onlsv/dosboxstaging_081_rc_has_now_been_released_with/

1. https://www.doomworld.com/forum/topic/142323-dosbox-staging-081-rc-has-now-been-released-with-3dfx-voodoo-1-support/
2. https://www.lemmingsforums.net/index.php?topic=6573.0
3. https://www.wcnews.com/chatzone/threads/dosbox-staging-0-81-rc-has-now-been-released-with-3dfx-voodoo-1-support.31811/ (Wing Commander)
4. https://www.cwaboard.co.uk/viewtopic.php?t=19736 (Carmageddon)
5. https://www.gog.com/forum/general/dosbox_staging_810

**Comment by johnnovak** [comment]
Great job everybody! 🚀 Now on to testing the RC in anger! 😎 Merry Christmas! 🎅🏻

**Comment by Grounded0** [comment]
Any ideas when the RC2 is released?

**Comment by johnnovak** [comment]
> Any ideas when the RC2 should be released?

Dunno. Maybe mid-Jan?

**Comment by Grounded0** [comment]
I have few comments to make on the release notes:

`config -wcd` seems to work fine when upgrading from 0.80.1 so we probably should mention that as an option when upgrading. Throwing away a 1,000 line .conf file and starting anew is not something many people will be happy about.

Starting without knowledge how Voodoo stuff works i had absolutely no idea where to place `glide2x.ovl` so where it should be placed should be mentioned. It also should be mentioned that it should be from Glide 2.43 driver distribution.

**Comment by johnnovak** [comment]
> `config -wcd` seems to work fine when upgrading from 0.80.1 so we probably should mention that as an option when upgrading. Throwing away a 1,000 line .conf file and starting anew is not something many people will be happy about.

It's a not good approach as it breaks you config. It does not work fine for all possible configs. We have moved properties around betweeon 0.80.1 and 0.81.0, those settings would break.

They don't need to throw away anything. They write the new clean config, then they carry over their old settings as needing. Usually that's just a few lines, surely they know what they configured in the global config as that's power user territory.

I'm not comfortable advertising a dodgy method of upgrading configs as that's just inviting trouble. I want to teach people how to do this correctly.

**Comment by Grounded0** [comment]
I'm going to deploy in Archive.org on 0.81.0 launch day a DOSBox Starter Pack:

```
$ ls -R ~/Desktop/dosboxstarterpack/
/Users/ajp/Desktop/dosboxstarterpack/:
misc  mt32-roms  soundfonts  voodoo  windows

/Users/ajp/Desktop/dosboxstarterpack/misc:
DEVICE.COM  EDIT.COM  EDIT.HLP

/Users/ajp/Desktop/dosboxstarterpack/mt32-roms:
CM32L_CONTROL.ROM  CM32L_PCM.ROM  MT32_CONTROL.ROM  MT32_PCM.ROM

/Users/ajp/Desktop/dosboxstarterpack/soundfonts:
4GMGSMT.sf2

/Users/ajp/Desktop/dosboxstarterpack/voodoo:
glide2x.ovl

/Users/ajp/Desktop/dosboxstarterpack/windows:
FAKESHAR.COM  S3_964  SB16DRV

/Users/ajp/Desktop/dosboxstarterpack/windows/S3_964:
8514FIX.FO_   CGA80WOA.FON  EGAMONO.LGO   ROMAN.FON     SMALLE.FON   TMSRF.FON     VGAOEM.FON
8514OEM.FO_   COURE.FON     EGAMONO.RLE   S3911GRB.3GR  SMALLF.FON   VDDS3VSN.386  VGASYS.FON
8514SYS.FO_   COURF.FON     HELVE.FON     S3VSNV.DRV    SSERIFE.FON  VERSION
928E_BT.EXE   EGA40850.FON  HELVF.FON     S3VSNV4.DRV   SSERIFF.FON  VGACOLOR.2GR
CGA40850.FON  EGA40WOA.FON  MODERN.FON    SCRIPT.FON    SYMBOLE.FON  VGAFIX.FON
CGA40WOA.FON  EGA80850.FON  OEMSETUP.INF  SERIFE.FON    SYMBOLF.FON  VGALOGO.LGO
CGA80850.FON  EGA80WOA.FON  RESTORE.TXT   SERIFF.FON    TMSRE.FON    VGALOGO.RLE

/Users/ajp/Desktop/dosboxstarterpack/windows/SB16DRV:
CHIPSET.PVL  CQA.TXT     DISK.ID       INSTALL.EXE  README.COM    SAMPLES2.PVL  WSWG.PVL
CMD.PVL      CSP.PVL     DRIVER.PVL    INSTALL.PVL  README.TXT    WINAPPL.PVL
CQA.COM      CTWAVE.PVL  FILELIST.PVL  PROGRAM.PVL  SAMPLES1.PVL  WINDRV.PVL
```

Just the basics to get everyone going in order to cut down the amount of unnecessary workload when people load crappy soundfonts and come whining here about it and so on.

**Comment by johnnovak** [comment]
> I'm going to deploy in Archive.org on 0.81.0 launch day a DOSBox Staging Starter Pack.

Good idea. My pack will come much later so you shouldn't wait for that.

A few suggestions:

- You might want to simply include the MT-32 ROMs from my pack. Filenames don't matter, we identify MT-32 ROMs by checksums now. https://archive.org/details/Roland-MT-32-ROMs
- GeneralUser GS is a quite nice SF2 file too. But even if you wanted to include this, I'd stop at these two.

**Comment by johnnovak** [comment]
> * GeneralUser GS is a quite nice SF2 file too. But even if you wanted to include this, I'd stop at these two.

4GMGSMT.sf2 is the best choice for a single set-and-forget SF2 file, agreed.

**Comment by Grounded0** [comment]
![rc2](https://github.com/dosbox-staging/dosbox-staging/assets/48192273/735de539-d6a9-4811-8ab1-27427b41756a)

**Comment by johnnovak** [comment]
Next week 🐶

**Comment by johnnovak** [comment]
@Grounded0 Just to keep you in the loop: we're having difficulties getting the macOS release built now with kcgen's MacMini being retired. This will take a while as it's only me and @kklobe who can do this, I have very little experience with the whole CI build machinery, and we're both busy.

It is what it is 🤷🏻

**Comment by Grounded0** [comment]
I'm not happy about these last moment surprises when we agreed to keep the build environment locked down until release but you just have to adapt for this stuff and make the best out of it.

**Comment by johnnovak** [comment]
> I'm not happy about these last moment surprises when we agreed to keep the build environment locked down until release but you just have to adapt for this stuff and make the best out of it.

Real-life issues trump our happiness I guess 😄 But we're past it now 👍🏻

**Comment by johnnovak** [comment]
Update: I found a [critical auto-shader regression ](https://github.com/dosbox-staging/dosbox-staging/pull/3330)(both kcgen and I myself missed it in, but the fix is a single character, fortunately), and @Burrito78 reported the app bundle crashed on Catalina. 

We'll continue tomorrow.

**Comment by Grounded0** [comment]
I think we can go for release before mid February if our issue tracker remains Dullsville (SimCity 2000 reference) it currently is.

**Comment by johnnovak** [comment]
> I think we can go for release before mid February if our issue tracker remains Dullsville (SimCity 2000 reference) it currently is.

I agree, we should drag it out if there isn't much movement. I'm only aware of a single show-stopper, the cursor issue on Win 11: https://github.com/dosbox-staging/dosbox-staging/issues/3345

But that's probably solvable by downgrading SDL for the Windows build if the SDL guys won't fix it in a week or two. Maybe that's the better solution because even they fixed this in a new version, who know what regressions they will introduce in that one...

**Comment by FeralChild64** [comment]
@johnnovak It looks like the SDL team is about to release SDL 2.30, with quite a lot of improvements, and our issue is scheduled to be a part of it - https://github.com/libsdl-org/SDL/milestone/9

**Comment by MasterO2** [comment]
> @johnnovak It looks like the SDL team is about to release SDL 2.30, with quite a lot of improvements, and our issue is scheduled to be a part of it - https://github.com/libsdl-org/SDL/milestone/9

So you're saying the Dosbox-Staging 0.81 full release should wait for the SDL team to release SDL version 2.30?

**Comment by FeralChild64** [comment]
I'm not sure. It might be safer to downgrade to 2.28.0.

**Comment by Grounded0** [comment]
I'd pick SDL 2.26.5 as the robust release for Windows. We want maximum robustness and not being constantly being shot in the foot by SDL. In the future we must be much more careful when bumping up SDL before releasing because its been consistent troublemaker with new releases bringing very little to the table.

**Comment by johnnovak** [comment]
> > I'm not sure. It might be safer to downgrade to 2.28.0.
> 
> I'd pick SDL 2.26.5 as the robust release for Windows. We want maximum robustness and not constantly being shot in the foot by SDL. In the future we must be much more careful when bumping up SDL before releasing because its been consistent troublemaker while new releases bring very little to the table.
> 
> There's a reason why SDL 2.28.0 got bumped to SDL 2.28.5 thus we should avoid it because it has other time bombs hidden there:
> 
> [Releases · libsdl-org/SDL](https://github.com/libsdl-org/SDL/releases)

I very much agree with that. SDL is a huge library with a vast API surface. It's entirely possible to use an older version for years without upgrading if the actual features you're using and the _way_ you are using it in your app are bug-free. Or buggy but you have reliable workarounds in place.

I've said this many times, we should just pin the version for the macOS and Windows builds and upgrade maybe once a year in a controlled manner. I don't think we've gained anything significant from tge SDL upgrades in the last 2 years, apart from high DPI. But the high DPI features tend to regress and break between minor releases.

On Linux we must keep it a moving target because of the idiosyncracies of Linux (repo maintainers packaging your stuff with a random SDL version, strong preference for dynamic linking, etc.) But I'll leave that to our Linux-first team members to decide.

I'm not saying never to upgrade SDL, but do so in a controlled manner. I hate it when people pull out the rug from under my feet...

Also, many of the SDL fixes and regressions are platform specific. It makes a lot more sense therefore to pin versions _per platform_.

**Comment by weirddan455** [comment]
We really just need to get the SDL2 wrap that @kklobe introduced working on all platforms.  Then we can pin a version using that for CI + binary releases (and keep the option of dynamic linking available for developers and Linux packagers).

I think Windows MSYS2 may end up being the most problematic though.  It's essentially a rolling release distro where all packages (including libraries, compilers, build tools, etc.) update at a moment's notice.

**Comment by johnnovak** [comment]
> We really just need to get the SDL2 wrap that @kklobe introduced working on all platforms. Then we can pin a version using that for CI + binary releases (and keep the option of dynamic linking available for developers and Linux packagers).

Yeah, I don't think it should be too hard.

> I think Windows MSYS2 may end up being the most problematic though. It's essentially a rolling release distro where all packages (including libraries, compilers, build tools, etc.) update at a moment's notice.

Are you serious? That's just unbelievably crap... What if someone needs to build against a specific old library version? That's another good point for ditching MSYS2 and only supporting MSVC on Windows.

**Comment by shermp** [comment]
If we move forward with using MSVC as the primary Windows builds, then someone needs to validate that the current git version of libslirp works correctly. I don't know enough about dosbox networking to know what to test to get as extensive coverage as possible.

**Comment by johnnovak** [comment]
> If we move forward with using MSVC as the primary Windows builds, then someone needs to validate that the current git version of libslirp works correctly. I don't know enough about dosbox networking to know what to test to get as extensive coverage as possible.

I don't know how to do that either. I don't have experience with DOSBox networking, neither too much interest in it 😅 Ideally, we'd need someone who's a developer and is into multi-player stuff over a network.

Related:
https://github.com/microsoft/vcpkg/issues/24604

**Comment by Grounded0** [comment]
MSYS2 keeps an archive of older builds too. SDL2 goes back to 2.0.22 so lets use this:

https://repo.msys2.org/mingw/x86_64/

**Comment by shermp** [comment]
> > If we move forward with using MSVC as the primary Windows builds, then someone needs to validate that the current git version of libslirp works correctly. I don't know enough about dosbox networking to know what to test to get as extensive coverage as possible.
> 
> I don't know how to do that either. I don't have experience with DOSBox networking, neither too much interest in it 😅 Ideally, we'd need someone who's a developer and is into multi-player stuff over a network.
> 
> Related: [microsoft/vcpkg#24604](https://github.com/microsoft/vcpkg/issues/24604)

Also related, my experimental branch adding the vcpkg overlay port: https://github.com/dosbox-staging/dosbox-staging/tree/sp/msvc-slirp-1



> MSYS2 keeps an archive of older builds too. SDL2 goes back to 2.0.22 so lets use this:
> 
> https://repo.msys2.org/mingw/x86_64/

Also an idea, although not officially supported by MSYS2. And they offer no guarantee of old packages sticking around. I suspect they start dropping old packages if they start running out of storage.

**Comment by Grounded0** [comment]
MSVC also doesn't support utilising the x87 so there's that.

**Comment by johnnovak** [comment]
> MSYS2 keeps an archive of older builds too. SDL2 goes back to 2.0.22 so lets use this:
> 
> https://repo.msys2.org/mingw/x86_64/

That's not a good idea, there was a lot of high-DPI related stuff going on between 2.0.22 and 2.26.5.

2.26.5 seems to be a safe choice, though.

> We already fired a bullet through MSVC's skull so i'm not in a mood to start resuscitating it.

Kinda by necessity because the MSCV pipeline broke in December. Since then the dev team members had a chat about this, and going with one build pipeline per platform seems to be an attractive and simple choice. macOS – XCode / clang, Windows - MSCV, Linux - gcc / clang.

We'll see what we decide.

**Comment by johnnovak** [comment]
> MSVC also doesn't support utilising the x87 so there's that.

Ah man... 😩 Screw this shit:
https://stackoverflow.com/a/77746210

**Comment by Grounded0** [comment]
```
mingw-w64-x86_64-SDL2-2.0.22-1-any.pkg.tar.zst     26-Apr-2022 07:39      1M
mingw-w64-x86_64-SDL2-2.0.22-1-any.pkg.tar.zst.sig 27-Apr-2022 16:35     566
mingw-w64-x86_64-SDL2-2.0.22-2-any.pkg.tar.zst     24-May-2022 20:04      1M
mingw-w64-x86_64-SDL2-2.0.22-2-any.pkg.tar.zst.sig 26-May-2022 18:02     566
mingw-w64-x86_64-SDL2-2.24.0-1-any.pkg.tar.zst     25-Aug-2022 13:22      1M
mingw-w64-x86_64-SDL2-2.24.0-1-any.pkg.tar.zst.sig 26-Aug-2022 19:11     566
mingw-w64-x86_64-SDL2-2.24.1-1-any.pkg.tar.zst     14-Oct-2022 17:03      1M
mingw-w64-x86_64-SDL2-2.24.1-1-any.pkg.tar.zst.sig 16-Oct-2022 08:15     566
mingw-w64-x86_64-SDL2-2.24.2-1-any.pkg.tar.zst     06-Nov-2022 04:00      1M
mingw-w64-x86_64-SDL2-2.24.2-1-any.pkg.tar.zst.sig 06-Nov-2022 13:59     566
mingw-w64-x86_64-SDL2-2.26.0-1-any.pkg.tar.zst     24-Nov-2022 03:43      1M
mingw-w64-x86_64-SDL2-2.26.0-1-any.pkg.tar.zst.sig 24-Nov-2022 06:39     566
mingw-w64-x86_64-SDL2-2.26.1-1-any.pkg.tar.zst     10-Dec-2022 19:04      1M
mingw-w64-x86_64-SDL2-2.26.1-1-any.pkg.tar.zst.sig 10-Dec-2022 21:44     566
mingw-w64-x86_64-SDL2-2.26.2-1-any.pkg.tar.zst     10-Jan-2023 17:00      2M
mingw-w64-x86_64-SDL2-2.26.2-1-any.pkg.tar.zst.sig 11-Jan-2023 08:04     566
mingw-w64-x86_64-SDL2-2.26.3-1-any.pkg.tar.zst     17-Feb-2023 16:04      1M
mingw-w64-x86_64-SDL2-2.26.3-1-any.pkg.tar.zst.sig 18-Feb-2023 11:38     566
mingw-w64-x86_64-SDL2-2.26.4-1-any.pkg.tar.zst     10-Mar-2023 16:01      1M
mingw-w64-x86_64-SDL2-2.26.4-1-any.pkg.tar.zst.sig 10-Mar-2023 18:18     566
mingw-w64-x86_64-SDL2-2.26.5-1-any.pkg.tar.zst     10-Apr-2023 09:11      1M
mingw-w64-x86_64-SDL2-2.26.5-1-any.pkg.tar.zst.sig 10-Apr-2023 17:14     566
mingw-w64-x86_64-SDL2-2.28.0-1-any.pkg.tar.zst     20-Jun-2023 21:09      1M
mingw-w64-x86_64-SDL2-2.28.0-1-any.pkg.tar.zst.sig 22-Jun-2023 11:24     566
mingw-w64-x86_64-SDL2-2.28.1-1-any.pkg.tar.zst     03-Jul-2023 21:07      1M
mingw-w64-x86_64-SDL2-2.28.1-1-any.pkg.tar.zst.sig 04-Jul-2023 06:29     566
mingw-w64-x86_64-SDL2-2.28.2-1-any.pkg.tar.zst     05-Aug-2023 15:25      1M
mingw-w64-x86_64-SDL2-2.28.2-1-any.pkg.tar.zst.sig 05-Aug-2023 18:05     566
mingw-w64-x86_64-SDL2-2.28.3-1-any.pkg.tar.zst     05-Sep-2023 09:11      1M
mingw-w64-x86_64-SDL2-2.28.3-1-any.pkg.tar.zst.sig 06-Sep-2023 17:46     566
mingw-w64-x86_64-SDL2-2.28.4-1-any.pkg.tar.zst     03-Oct-2023 09:12      1M
mingw-w64-x86_64-SDL2-2.28.4-1-any.pkg.tar.zst.sig 04-Oct-2023 06:36     566
mingw-w64-x86_64-SDL2-2.28.5-1-any.pkg.tar.zst     03-Nov-2023 09:07      1M
mingw-w64-x86_64-SDL2-2.28.5-1-any.pkg.tar.zst.sig 03-Nov-2023 13:48     566
```

**Comment by Grounded0** [comment]
Ship with SDL 2.26.5 and lets think about future after that.

**Comment by johnnovak** [comment]
Yeah, it's unfortunate we did not do this pre-RC1. Maybe we'll have a shortlived RC3, depending on many more issues will be found.

**Comment by weirddan455** [comment]
Well, we might run into issues just grabbing an old MSYS2 package.  Doing so it basically a partial upgrade and is unsupported.  You'll run into problems if (for example) SDL was built against version 1 of some dependency.  Then that dependency gets upgraded to version 2 by MSYS2.

>  After you have installed MSYS2 via the installer or other means, you need to continue updating it with the built-in pacman tool. MSYS2 is a [rolling release](https://en.wikipedia.org/wiki/Rolling_release) distribution and only supports full system upgrades, which means there are frequent minor and major updates to various packages and you can only update all packages at once.

https://www.msys2.org/docs/updating/

If we're just looking for an SDL binary to ship, I'd probably grab one straight from the SDL project (https://github.com/libsdl-org/SDL/releases).  I know at least with the MSVC build of SDL, they make it a point to not dynamically link with anything so it's not a problem (they don't even link with the C runtime library).

**Comment by johnnovak** [comment]
> If we're just looking for an SDL binary to ship, I'd probably grab one straight from the SDL project (https://github.com/libsdl-org/SDL/releases). I know at least with the MSVC build of SDL, they make it a point to not dynamically link with anything so it's not a problem (they don't even link with the C runtime library).

But wait a minute, I just remembered, I think this is just a matter of downgrading our `sdl2.wrap` version in `subprojects` and use different versions per platform. So fortunately which version of SDL2 MSYS2 is on doesn't even come into the picture.

**Comment by japsmits** [comment]
> > MSVC also doesn't support utilising the x87 so there's that.
> 
> Ah man... 😩 Screw this shit: https://stackoverflow.com/a/77746210

Major problem here is that AMD decided to drop support for the FP unit (which had 8 80-bit float values) in favor of SSE (which "only" has 64-bit precision) with their move to x64. GCC and Intel compilers only support 128-bit calculations if the platform supports it. Even in C++23, 128-bit precision is optional (although standardized).

Only option would be to go the soft-float way. There is bigfloat (https://github.com/ifd3f/bigfloat) for example. And probably others as well, but this is what I found in a 5-minute search. Won't speed things up, but if precision is needed that might just be worth it.

**Comment by weirddan455** [comment]
>  Major problem here is that AMD decided to drop support for the FP unit (which had 8 80-bit float values) in favor of SSE (which "only" has 64-bit precision) with their move to x64.

Where are you getting this information from?  AMD processors (as well as Intel) are still backwards compatible.  I think most modern computers still start up in real mode and you can run FreeDOS on them natively.

I've read (very recently) about a push to drop some of this legacy cruft from x64 but I don't think any hardware has been shipped that does this yet.

**Comment by kklobe** [comment]
> > Major problem here is that AMD decided to drop support for the FP unit (which had 8 80-bit float values) in favor of SSE (which "only" has 64-bit precision) with their move to x64.
> 
> Where are you getting this information from? AMD processors (as well as Intel) are still backwards compatible. I think most modern computers still start up in real mode and you can run FreeDOS on them natively.
> 

Yep, AMD processors definitely support x87 as can be seen in the [AMD64 Architecture Programmer’s Manual, Volume 5: 64-Bit Media and x87 Floating-Point Instructions](https://www.amd.com/content/dam/amd/en/documents/processor-tech-docs/programmer-references/26569.pdf)

> I've read (very recently) about a push to drop some of this legacy cruft from x64 but I don't think any hardware has been shipped that does this yet.

Agree, I think we're safe for the foreseeable future, the latest Intel X86S proposal doesn't mention removing x87, and even documents initializing the x87 status/control/tag words: https://www.intel.com/content/www/us/en/developer/articles/technical/envisioning-future-simplified-architecture.html

**Comment by Grounded0** [comment]
![image](https://github.com/dosbox-staging/dosbox-staging/assets/48192273/833857c8-bba6-48c8-abf6-8edca6b4063f)

**Comment by japsmits** [comment]
Hmm, I should do more fact checking before and not rely on memory too much, apparently. I stand corrected.

**Comment by Grounded0** [comment]
Does mid February sound like a realistic ship date? Valentine's Day is on the 14th.

**Comment by johnnovak** [comment]
> Does mid February sound like a realistic ship date?

Yeah looks like the code is stable enough now for the final, and we fixed our SDL situation for the Windows build.

I'm working on website improvements, the I'll need to update the release notes for the final, plus I think I'll make one or two "marketing materials" (images) too. 

So sometimes before the end of next week seems realistic.

I'll post links to the finals builds here for internal testing in a few days.

> Valentine's Day is on the 14th.

Good to know 😆

**Comment by johnnovak** [comment]
@Grounded0 

The final deliverables if you wanna test:
https://drive.google.com/drive/folders/1KLQ40FK9BnJWrGrDp3dfqoaDZ7Mp85t0?usp=sharing

The core team is happy with them.

I'll update the detailed PR list of the release notes tomorrow, then release on Wednesday.

---

I've also prepared some "marketing materials", feel free to spam the whole world with these images once the release is out.

Suggested accompanying text:

1. "The griffin has spoken."
2. "Education is important."

 😎 🤘🏻 

![crt-shaders](https://github.com/dosbox-staging/dosbox-staging/assets/698770/73d52c66-6cd0-4647-a039-35452f902e69)

![aspect-ratio](https://github.com/dosbox-staging/dosbox-staging/assets/698770/989d8592-baad-4a02-9bf4-1856c8f7e64b)

**Comment by Grounded0** [comment]
Valentine's Day reference would be welcome.

Thanks to everyone who made DOSBox Staging 0.81.0 a reality.

**Comment by johnnovak** [comment]
> Valentine's Day reference would be welcome.

I don't do Valentine's Day 😎 🤷🏻 

> Thanks to everyone who made DOSBox Staging 0.81.0 a reality.

![image](https://github.com/dosbox-staging/dosbox-staging/assets/698770/91e737fe-2657-4176-b801-22395c6a229c)

**Comment by Grounded0** [comment]
![image](https://github.com/dosbox-staging/dosbox-staging/assets/48192273/e6a91f24-f4db-45a7-9cf2-5c8f4a5631e9)

**Comment by Grounded0** [comment]
We'll probably need to fix this:

```
07:28Z <  %Anonymised> the changelog indicates that one will be written when the new version is started
07:29Z <  Grounded0> no we definitely will not nuke your configs
07:30Z <  %Anonymised> "After backing up the existing primary config, simply start the new version—a new dosbox-staging.conf will be written containing the new defaults and updated setting  descriptions."
07:30Z <  %Anonymised> which is the false statement
```

**Comment by johnnovak** [comment]
> We'll probably need to fix or clarify this:
> 
> ```
> 07:28Z <  %Anonymised> the changelog indicates that one will
> be written when the new version is started
> 07:29Z <  Grounded0> no we definitely will not nuke your configs
> 07:30Z <  %Anonymised> "After backing up the existing primary
> config, simply start the new version—a new dosbox-staging.conf
> will be written containing the new defaults and updated setting  
> descriptions."
> 07:30Z <  %Anonymised> which is the false statement
> ```

The instructions are correct. The guy is confused.

**Comment by Grounded0** [comment]
I'll deliver the DOSBox Staging Starter Pack on Thursday as i promised and post Archive.org link here.

**Comment by Grounded0** [comment]
DOSBox Staging Starter Pack:

https://archive.org/details/dosboxstarterpack

**Comment by johnnovak** [comment]
> DOSBox Staging Starter Pack:
> 
> https://archive.org/details/dosboxstarterpack

Good stuff, man!

**Comment by Grounded0** [comment]
I'm going to close this ticket on Sunday.

**Comment by Grounded0** [comment]
I'm pushing in 200 USD per month for the next 3 months so anyone who fixes a 0.81.x bug should be getting their cut.

**Comment by johnnovak** [comment]
> I'm pushing in 200 USD per month for the next 3 months so anyone who fixes an 0.81.x bug should be getting a reward.

@Grounded0 More donations are always welcome, don't get me wrong, and we're grateful for your donations to the project so far. I'm just a bit wary that this would blur the lines between donations and bounties, as donations are supposed to be "no strings attached".  

How about offering bounties instead on individual issues tickets that are important to you? I'm just suggesting this to set clear expectations on all sides and not blur the lines.

**Comment by Grounded0** [comment]
I think it would be a great idea to target donations to bug fixers now in order to improve release quality. I don't have admin access to Open Collective so it's not my call.

**Comment by johnnovak** [comment]
Maybe in the future we could try to do a "backlog issue fixes only" release. But then, it depends on what people wanna do. We work on what interests us.

**Comment by QmwJlHuSg9pa** [comment]
> DOSBox Staging Starter Pack:
> 
> https://archive.org/details/dosboxstarterpack

Hi, regarding the inclusion of Glide v2.43 -- I just wanted to refute the following assertion made in a pull request thread (https://github.com/dosbox-staging/dosbox-staging/pull/3103#issuecomment-1802917258):

> > perhaps this version 2.43 should be tested with the games first to see if it causes crashes
> 
> I've tested it 2.43 specifically with all the DOS Voodoo games and demos, and confirmed there are no issues. These are screenshots using 2.43:
> [...]

I have noticed some profound issues during my testing of Carmageddon (I have not yet tested any other games).

There seems to be some sort of off-by-one issue when rendering HUD elements, as well as the 3D viewport, when compared to the Glide driver included with the GOG.com release (version unknown). On the other hand, v2.43 seems to render transparent shadows, which is nice.

Compare the following images -- GOG Glide, followed by Glide v2.43 (note the crusty HUD and the black line running down the right side of the viewport):
![gog_glide_before](https://github.com/dosbox-staging/dosbox-staging/assets/1274826/08c35c70-5bf6-4517-96f1-65c916a4581e)
![glide243_before](https://github.com/dosbox-staging/dosbox-staging/assets/1274826/367a7d51-9980-42fd-a4cb-5ab3484b9c0b)

Now, here's Glide v2.43 when switching to cockpit view then back again (note the junk on the right edge of the viewport):
![glide243_cockpit](https://github.com/dosbox-staging/dosbox-staging/assets/1274826/2bef8fbf-0abb-4e32-b905-41e9ec05655c)
![glide243_after](https://github.com/dosbox-staging/dosbox-staging/assets/1274826/45a33c5c-bc69-409d-ba99-f21383a5308b)

**Comment by johnnovak** [comment]
@QmwJlHuSg9pa This is not the place for such discussions. If you have good reason to believe there is a problem with the Voodoo emulation, please raise a separate issue ticket with all relevant details.

But no "what if" scenarios please. If you don't know how the game is supposed to look on real hardware, maybe it's best to leave it at that.

**Comment by QmwJlHuSg9pa** [comment]
> @QmwJlHuSg9pa This is not the place for such discussions. If you have good reason to believe there is a problem with the Voodoo emulation, please raise a separate issue ticket with all relevant details.
> 
> But no "what if" scenarios please. If you don't know how the game is supposed to look on real hardware, maybe it's best to leave it at that.
> 
> Comparing things to the results you're getting with Glide, which is a high-level approximation of Voodoo, is rather pointless.

I have no opinion on the current state of emulation accuracy. Just pointing out that a main assertion that lead to the promotion of v2.43 on your Wiki was somewhat baseless in the first place. In the linked pull request thread, there's clear graphical corruption in multiple screenshots of various games, similar to my own descriptions and screenshots.

To counter what Grounded0 stated after the screenshots were posted:

> At least we managed to hunt down that good Glide 2.43 DOS driver. This can now progress.

Sorry, but it doesn't seem to hold that v2.43 is good in all instances.

**Comment by Grounded0** [comment]
Open separate ticket for any Voodoo issue please.

**Comment by QmwJlHuSg9pa** [comment]
> GOG is using either Glide wrapper or OpenGL backend for the Voodoo patch which is omitted from DOSBox Staging. We do pure software rendering. Please open a separate ticket for any Voodoo issues.

Thanks for pointing this out; it would have taken some time before this became apparent to me, as the GOG workaround is undocumented. The GOG version is likely utilising a build of openglide (but in my case is perhaps falling back to software mode?). FWIW, I tested the build of `glide2x.ovl` included with the USA re-release CD and it appears virtually identical to v2.43.

Anyway, probably something worth mentioning in the Wiki for anyone else in the future looking to adopt v2.43, so they may avoid scratching their head at the outcomes?

Not sure if there's any point opening a ticket, as it looks to be a similar issue to #2943.

**Comment by johnnovak** [comment]
> Anyway, probably something worth mentioning in the Wiki for anyone else in the future looking to adopt v2.43, so that they may avoid scratching their head at the outcomes?

Please write a succinct description of what should go into the wiki and/or what should be changed and I'll paste it in. I just don't want to understand all these details right now 😄 

It's fine to post your wiki changes draft here in a comment **as an exception** now, but please in the future always either raise separate issue or discussion ticket for these things, instead of overloading vaguely related other tickets. Otherwise chaos ensues...

**Comment by Grounded0** [comment]
When is 0.81.1 going to ship?

**Comment by johnnovak** [comment]
> When is 0.81.1 going to ship?

We could ship it now, and I wanted to release it this week. We started testing the build internally, then the Tyrian keyboard regression came in, so I want to wait for a fix for that. FeralChild is away for a week or so, we probably need him to fix it.

**Comment by Grounded0** [comment]
Donated another 200 USD. Hopefully i get that 0.81.1 soon.

**Comment by johnnovak** [comment]
> Donated another 200 USD. Hopefully i get that 0.81.1 soon.

Technically, we could release now, but we thought with FeralChild that it's worthwhile to test this last fix a big more thoroughly:
https://github.com/dosbox-staging/dosbox-staging/pull/3647

**Comment by Grounded0** [comment]
Yeah i said that bug will be fixed for 0.81.1 in `#dosgameclub` so lets make sure we get it right.

**Comment by FeralChild64** [comment]
Regarding the https://github.com/dosbox-staging/dosbox-staging/pull/3647 - I still don’t have a test program which could be run on the real hardware. It seems the IRQ request is a PIC internal value, and there is probably no way to read it. I also don’t know yet how to takeover the interrupt on the guest side without using the z86 assembler.

**Comment by johnnovak** [comment]
@Grounded0 Tomorrow we'll release v0.81.1.
